test/test_utils/sagemaker.py

```diff
@@ -240,25 +243,34 @@ def generate_sagemaker_pytest_cmd(image, sagemaker_test_type):
     )
 
 
-def install_sm_local_dependencies(framework, job_type, image, ec2_conn, ec2_instance_ami):
+@retry(reraise=True, wait=wait_fixed(60), stop=stop_after_attempt(5))
+def install_python_in_instance(context, python_version="3.9"):
     """
     Install sagemaker local test dependencies
-    :param framework: str
-    :param job_type: str
-    :param image: str
-    :param ec2_conn: Fabric_obj
+    :param context: Invoke Context / Fabric Connection object
+    :param python_version: str python version to install, such as 3.8, 3.9, etc.
     :return: None
     """
-    python_invoker = get_python_invoker(ec2_instance_ami)
-    # Install custom packages which need to be latest version"
-    # using virtualenv to avoid package conflicts with the current packages
-    ec2_conn.run(f"sudo apt-get install virtualenv python3-pip -y ")
-    ec2_conn.run(f"virtualenv env --python {python_invoker}")
-    ec2_conn.run(f"source ./env/bin/activate")
-    if framework == "pytorch":
-        # The following distutils package conflict with test dependencies
-        ec2_conn.run("sudo apt-get remove python3-scipy python3-yaml -y")
-    ec2_conn.run(f"sudo {python_invoker} -m pip install -r requirements.txt ", warn=True)
+    context.run("""ls ~/.pyenv || git clone https://github.com/pyenv/pyenv.git ~/.pyenv""")
+    # Need to configure PATH and PYENV_ROOT changes in alternative location because ~/.bashrc and ~/.profile are
+    # not used when using bash through Fabric Connection.
+    context.run("sudo chmod 666 /etc/profile.d/dlami.sh")
+    context.run("""echo 'export PYENV_ROOT="$HOME/.pyenv"' >> /etc/profile.d/dlami.sh""")
+    context.run(
+        """echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> /etc/profile.d/dlami.sh"""
+    )
+    context.run("""echo 'eval "$(pyenv init -)"' >> /etc/profile.d/dlami.sh""")
+    context.run("sudo chmod 644 /etc/profile.d/dlami.sh")
+
+    context.run("sudo apt-get update")
+    context.run(
+        "sudo apt install -y make build-essential libssl-dev zlib1g-dev "
+        "libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm "
+        "libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev"
+    )
+
+    context.run(f"pyenv install {python_version}")
+    context.run(f"pyenv global {python_version}")
```

Comment:
Should we add a validation step here to verify our pretest setup was successful?

Reply:
Makes sense. Will do.